chunker/chunk_test.go

```diff
@@ -254,44 +257,80 @@ func TestFileReader(t *testing.T) {
 		require.NoError(t, err)
 		expectedOutcomes[i] = data.content
 	}
-
 	files, err := ioutil.ReadDir(testFilesDir)
 	require.NoError(t, err)
 
 	for i, file := range files {
-
 		testfilename := filepath.Join(testFilesDir, file.Name())
 		reader, cleanup := FileReader(testfilename, nil)
-
 		bytes, err := ioutil.ReadAll(reader)
-
 		require.NoError(t, err)
 		contents := string(bytes)
 		//compare file content with correct string
 		require.Equal(t, contents, expectedOutcomes[i])
 		cleanup()
 	}
-
-	if err := os.RemoveAll(dir); err != nil {
-		t.Fatalf("Error removing direcotory: %s", err.Error())
-	}
-
 }
 
 func TestDataFormat(t *testing.T) {
 	_, thisFile, _, _ := runtime.Caller(0)
 	dir := "test-files"
 	require.NoError(t, os.MkdirAll(dir, os.ModePerm))
+	defer deleteDirs(t, dir)
 	testFilesDir := filepath.Join(filepath.Dir(thisFile), "test-files")
 	expectedOutcomes := [5]InputFormat{2, 1, 0, 2, 1}
-
 	file_data := [5]string{"test-1.json", "test-2.rdf", "test-3.txt", "test-4.json.gz", "test-5.rdf.gz"}
+
 	for i, data := range file_data {
 		filePath := filepath.Join(testFilesDir, data)
 		format := DataFormat(filePath, "")
-
 		require.Equal(t, format, expectedOutcomes[i])
+	}
+}
+
+func TestRDFChunkerChunk(t *testing.T) {
+	_, thisFile, _, _ := runtime.Caller(0)
+	dir := "test-files"
+	require.NoError(t, os.MkdirAll(dir, os.ModePerm))
+	defer deleteDirs(t, dir)
+	testFilesDir := filepath.Join(filepath.Dir(thisFile), "test-files")
+	dataFile := filepath.Join(testFilesDir, "data.rdf")
+	resultData := filepath.Join(testFilesDir, "result.rdf")
+	//download data
+	cmd := exec.Command("wget", "-O", dataFile, "https://media.githubusercontent.com/media/dgraph-io/benchmarks/master/ldbc/sf0.3/ldbc_rdf_0.3/Deltas.rdf")
+
+	if out, err := cmd.CombinedOutput(); err != nil {
+		fmt.Printf("Error %v", err)
+		fmt.Printf("Output %v", out)
+	}
+	f, err := os.Create(resultData)
+	require.NoError(t, err)
+	chunker := NewChunker(RdfFormat, 1000)
+	rd, _ := FileReader(dataFile, nil)
+	chunkBuf, _ := chunker.Chunk(rd)
 
+	for chunkBuf.Len() > 0 {
+		str, err := chunkBuf.ReadString('\n')
+		require.NoError(t, err)
+		_, err = f.WriteString(str)
+		require.NoError(t, err)
+	}
+	if err != nil {
+		log.Fatal(err)
```

Comment:
we should fail the test instead of calling log.Fatal

Reply:
and everywhere below

Reply:
done